### PR TITLE
[Proposal] Add built-in function register_executing() and register_recording()

### PR DESCRIPTION
**Pull request by h-east**
Hi Bram and list,

I added a state of recording into register and/or executing the register to built-in function mode(1).
See the modified document in the [patch](https://github.com/vim/vim/compare/master...h-east:proposal_mode_function_improvement?expand=1#diff-bea881dfa9626bab7141337b0fcdb23e). 

How about this?
Possibly, should we add another function?

--
Best regards,
Hirohito Higashi (h_east)

### CODE REVIEW

**Comment by brammool** [comment]
Hirohito Higashi wrote:

> I added a state of recording into register and/or executing the register to built-in function mode(1).
> See the modified document in the [patch](https://github.com/vim/vim/compare/master...h-east:proposal_mode_function_improvement?expand=1#diff-bea881dfa9626bab7141337b0fcdb23e). 
> 
> How about this?
> Possibly, should we add another function?

Yeah, this is not part of the mode, this state doesn't change what
commands will do.

Perhaps recording() can return the register name and executing() can
return one or zero.

-- 
Microsoft is to software what McDonalds is to gourmet cooking

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by h-east** [comment]
I reverted `mode()` modification and added function `register_executing()` and `register_recording()`.

**Comment by brammool** [comment]
> I reverted `mode()` modification and added function
> `register_executing()` and `register_recording()`.

Looks OK, but the names are a bit long.  How about reg_executing() and
reg_recording()? That's clear enough, right?

We could make them event a bit short:
	reg_excute()
	reg_record()
But that sounds like actions, not a good choice.

-- 
hundred-and-one symptoms of being an internet addict:
48. You get a tatoo that says "This body best viewed with Netscape 3.1 or
    higher."

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by h-east** [comment]
Rename function name to `reg_executing()` and `reg_recording()` :+1:

File: runtime/doc/usr_41.txt [hunk 1]
```diff
@@ -1016,6 +1016,8 @@ Various:					*various-functions*
 	getreg()		get contents of a register
 	getregtype()		get type of a register
 	setreg()		set contents and type of a register
+	regster_executing()	return the state of the register executing
```
Comment (markonm) [inline runtime/doc/usr_41.txt]: s/regster/reg

File: runtime/doc/usr_41.txt [hunk 2]
```diff
@@ -1016,6 +1016,8 @@ Various:					*various-functions*
...
+	regster_recording()	return the state of the register recording
```
Comment (markonm) [inline runtime/doc/usr_41.txt]: s/regster/reg

**Comment by h-east** [comment]
@xtal8 Thanks! I fixed.

**Comment by andymass** [comment]
This is really neat!  Of course, the only thing missing now is an autocmd or two on "start recording" and "stop recording" :).

**Pull request closed without merging**